relay-server/src/services/processor.rs

```diff
@@ -1314,37 +1325,62 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState<G>,
     ) -> Result<(), ProcessingError> {
-        let options = &self.inner.global_config.current().options;
+        let attachment_type = state
+            .envelope()
+            .get_item_by(|item| item.attachment_type().is_some())
+            .and_then(|item| item.attachment_type())
+            .map(|ty| ty.to_string())
+            .unwrap_or("none".to_string());
+
+        if !state.has_event() {
+            metric!(
+                counter(RelayCounters::NormalizationDecision) += 1,
+                attachment_type = attachment_type.as_ref(),
+                decision = "no_event"
+            );
 
-        let full_normalization = if self.inner.config.processing_enabled()
-            && options.processing_disable_normalization
-            && (state.envelope().meta().is_from_internal_relay())
-        {
+            // NOTE(iker): only processing relays create events from
+            // attachments, so these events won't be normalized in
+            // non-processing relays even if the config is set to run full
+            // normalization.
             return Ok(());
-        } else if !self.inner.config.processing_enabled() && options.force_full_normalization {
-            true
-        } else {
-            match self.inner.config.normalization_level() {
-                NormalizationLevel::Disabled => {
-                    // We assume envelopes coming from an internal relay have
-                    // already been normalized. During incidents, like a PoP region
-                    // not being available, envelopes can go to other PoP regions or
-                    // directly to processing relays. Events should be fully
-                    // normalized, independently of the ingestion path.
-                    if self.inner.config.processing_enabled()
-                        && (!state.envelope().meta().is_from_internal_relay())
-                    {
-                        true
-                    } else {
-                        relay_log::trace!("Skipping event normalization");
-                        return Ok(());
-                    }
+        }
+
+        let options = &self.inner.global_config.current().options;
+        let from_internal = state.envelope().meta().is_from_internal_relay();
+
+        let full_normalization = match self.inner.config.normalization_level() {
```

Comment:
With the new item header, do we still need the `processing` check? Could we now simplify this to (with the suggestion above):

```rust
let full_normalization = match self.inner.config.normalization_level() => {
    NormalizationLevel::Full => true,
    NormalizationLevel::Default => ! state.event_fully_normalized.
};
```

Second question: Do we still need the global config options? If so, can we create a helper function that returns a normalization level depending on static + global config? E.g. `self.inner.config.normalization_level(options)`.

Reply:
What do we need `options.force_full_normalization` for?

I think the expression in the `Default` branch can be simplified to 

```rust
if self.inner.config.processing_enabled()
      && options.processing_disable_normalization
      && from_internal
      && state.event_fully_normalized
  {
      // ...
  } else {
      self.inner.config.processing_enabled() || options.force_full_normalization
  }
  ```

Reply:
There are two feature flags, and `options.force_full_normalization` is for PoP relays. Simplified the branch too!